python_modules/dagster/dagster/_core/definitions/decorators/assets_definition_factory.py

```diff
@@ -202,59 +205,115 @@ def make_keys_by_output_name(
     return {output_name: asset_key for asset_key, (output_name, _) in asset_outs.items()}
 
 
-class InOutMapper:
+def compute_required_resource_keys_for_underlying_op(
+    explicitly_passed_required_resource_keys: Set[str],
+    resource_defs_bound_to_asset: Mapping[str, ResourceDefinition],
+    fn: Callable[..., Any],
+) -> Set[str]:
+    arg_resource_keys = {arg.name for arg in get_resource_args(fn)}
+    check.param_invariant(
+        len(explicitly_passed_required_resource_keys) == 0 or len(arg_resource_keys) == 0,
+        "Cannot specify resource requirements in both @multi_asset decorator and as"
```

Comment:
This message assumes that this is being called from the "@multi_asset" decorator, but the naming and location of this function don't communicate that it's only meant to be called from that context.